_posts/2018-12-11-dipto-csc-799-report.md

```diff
@@ -0,0 +1,234 @@
+
+		if key=='sentences_tone':
+			tone_dict = {}
+			for sentence_idx in range(len(output['sentences_tone'])):
+				sentence = output['sentences_tone'][sentence_idx]
+				data = sentence['tones']
+				# pprint(data)
+				for entry in data:
+					if entry['tone_name'] not in tone_dict.keys():
+						tone_dict[entry['tone_name']] = entry['score']
+					else:
+						tone_dict[entry['tone_name']] += entry['score']
+			for tone_name in tone_names:
+				# print(tone_name)
+				if tone_name in tone_dict.keys():
+					score = tone_dict[tone_name]
+				else:
+					score = 0
+				row.append(score)
+			row.append(len(output['sentences_tone']))
+			row.append(cls)
+	print(len(row))
+	all_data.append(row)
+```
+
+In my research, I had text data of two different categories. I calculated the emotions, language, and social scores of those documents using IBM Tone Analyzer. I used these scores to see if there is any statistical differences among the scores of the documents of different classes using Student's t-test.
+
+# Student t-test
+
+Let me start by telling a story. There was a person named William Sealy Gosset. He worked at Guiness Brewery over one hundred years ago. He came up with a statistical test to show the difference between barley yield from two fields. When he wanted to publish the test, he was nervous, and instead of publishing it in his name, he used the pseudonym 'Student'. To this day, this test is known as Student's t-test instead of Gosset's t-test.
+
+# What is t-test?
+Imagine, you have two fields of same crops - field 1 and field 2. Maybe you want to compare the productions of these two fields with respect to a certain criteria. However, obviously it's not wise to cut the crops from the whole fields for this. A test on samples from both these fields should be enough. Look at the following image:
+
+![Different distributions from two fields](/images/ttest.jpg)
+
+*Image collected from: http://www.socialresearchmethods.net/kb/stat_t.htm*
+
+The top figure says that field 2 (indicated by blue line) has a higher mean than field 1 (indicated by green curve). But that just says a partial story. The fields can have different distributions as shown in the other two figures. Depending on that, the crops from these two fields can have statistical differences or not. And that's where t-test becomes useful. We can see the t-value as a ratio between signal and noise. Signal means the numbers that can show the differences in two distributions, if any, and noise means the numbers that are just outliers. We can define signal as difference between group means and noise is the variability in the groups.
+
+The formula that we are going to use:
+
+t-value = |mean of field 1 - mean of field 2|/(variance of field 1/sample count of field 1 + variance of field 2/sample count of field 2)^0.5
+
+A sample calculation is shown in the following google sheet: https://docs.google.com/spreadsheets/d/1H8ajhRR5SOqkcx3pBPHjA1LY76j8-mtNDsnGajFBrXM/edit?usp=sharing
```

Comment:
Instead of having the Google sheet link, turn that sheet into a HTML table. This should help:
https://www.tablesgenerator.com/markdown_tables